src/bootstrap/tosh.c

```diff
@@ -13,30 +13,74 @@
 #include <stdlib.h>
 #include <ft_printf.h>
 #include <unistd.h>
-#include "../error/error.h"
+
+#include "../env/env.h"
 #include "../input/input.h"
+#include "../lexer/lexer.h"
+#include "../parser/parser.h"
+#include "../error/error.h"
+
+static void		shell_flow_control(const char *const input, t_env *const env)
+{
+	t_list_meta					tokens;
+	struct s_complete_command	*complete_command;
+	bool						extra_input_requested;
+	t_error						err;
+
+	if (lexer_tokenize(&tokens, input) == -1)
+	{
+		ft_dprintf(STDERR_FILENO, "tosh: lexer could not allocate memory\n");
+	}
+	if (tokens.len != 0)
+	{
+		err = parser_parse(&complete_command, &extra_input_requested, &tokens);
+		lexer_clear(&tokens);
+		if (is_error(err))
+		{
+			ft_dprintf(STDERR_FILENO, "tosh: %s\n", err.msg);
+			return ;
+		}
+		if (extra_input_requested)
+		{
+			ft_dprintf(STDERR_FILENO, "extra input requested\n");
+			return ; //not sure how to hack this in yet
+		}
+		// exec_run(complete_command, env);
+		(void)env;
+		parser_del(&complete_command);
+	}
+}
 
 /*
 ** TODO: Consider using ft_getline if TERM is unknown.
 */
 
-void	tosh(void)
+void			tosh(char **envp)
 {
+	t_env				*env;
 	char				*input;
 	t_error				error;
 	char				prompt[32];
 
+	env = env_from_envp(envp);
+	if (env == NULL)
+	{
+		ft_dprintf(STDERR_FILENO, "tosh: fatal: unable to create enviroment\n");
+		exit(1);
+	}
 	ft_snprintf(prompt, sizeof(prompt), "%{green}TOSH $ %{reset}");
 	while (true)
 	{
 		error = input_read(&input,
-			(struct s_term_formatted_string){prompt, 7});
+			(struct s_term_formatted_string){prompt, 7});//why the arbitrary magic number on a dynamic variable len, pls fix
```

Comment:
can't we make these both be defines?
if the control codes can be different then why not the width?

Reply:
> can't we make these both be defines?

Sure, send a PR. Although I'm not sure it would be the best use of our time right now.

>  if the control codes can be different then why not the width?

I'm confused by what you mean. Both are always the same in this snippet.